### PR TITLE
Laboratory work #3, Lubov Gorbunova - 21FPL2

### PR DESCRIPTION
**Pull request by lubagorbunova**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -173,7 +204,17 @@ def extract_pairs(tokens: tuple[int, ...], window_length: int) -> Optional[tuple
...
     """
-    pass
+    if not tokens:
+        return None
+    if not isinstance(window_length, int) or window_length < 2:
+        return None
+    pairs = []
+    for item in range(len(tokens) - window_length + 1):
+        for i in range(window_length):
+            if tokens[item] != tokens[item + i]:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:214]: extract `tokens[item]` and `tokens[item + i]` before `if` and use as variables

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — COMMENTED** [review]
I believe you can do more!

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -51,7 +52,13 @@ def _clean_and_tokenize(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens
         """
-        pass
+        text_without_punctuation = ''
+        for item in text:
+            if item not in self.punctuation:
+                text_without_punctuation += item
+        lowercase_text = text_without_punctuation.lower()
+        tokens = tuple(lowercase_text.split())
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:60]: return without creating a variable

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -66,7 +73,11 @@ def _remove_stop_words(self, tokens: tuple[str, ...]) -> tuple[str, ...]:
             tuple[str, ...]
                 tokens without stop-words
         """
-        pass
+        result = []
+        for item in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:77]: do it in one line

File: lab_3_keywords_textrank/main.py [hunk 4]
```diff
@@ -136,7 +154,13 @@ def encode(self, tokens: tuple[str, ...]) -> Optional[tuple[int, ...]]:
                 sequence of integer tokens
         In case of empty tokens input data, None is returned
         """
-        pass
+        self._learn_indices(tokens=tokens)
+        if not tokens:
+            return None
+        integer_tokens = []
+        for item in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:161]: again - one line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (lubagorbunova) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 2 above)
Comment (lubagorbunova) [reply lab_3_keywords_textrank/main.py]: done

**Pull request closed without merging**